### ISSUE TITLE
hostname param not set

### ISSUE THREAD

**Issue by brightdroid** [joshuabaird/puppet-ipaclient#49]
When `/etc/hostname` contains a hostname (which is correct) and not a fqdn, `ipa-client-install` fails.

Fixed it by adding a hostname param which defaults to `$::fqdn`

```diff
diff --git a/manifests/init.pp b/manifests/init.pp
index 3b8878c..c681871 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -105,6 +105,7 @@ class ipaclient (
   $ssh                = $ipaclient::params::ssh,
   $sshd               = $ipaclient::params::sshd,
   $sudo               = $ipaclient::params::sudo,
+  $hostname           = $ipaclient::params::hostname,
 ) inherits ipaclient::params {
 
   package { $package:
@@ -188,10 +189,16 @@ class ipaclient (
         $opt_sudo = ''
       }
 
+      if $hostname {
+        $opt_hostname = ['--hostname', $hostname]
+      } else {
+        $opt_hostname = ''
+      }
+
       # Flatten the arrays, delete empty options, and shellquote everything
       $command = shellquote(delete(flatten([$installer,$opt_realm,$opt_password,
                             $opt_principal,$opt_mkhomedir,$opt_domain,
-                            $opt_server,$opt_fixed_primary,$opt_ssh,$opt_sshd,$opt_ntp,$opt_sudo,
+                            $opt_server,$opt_fixed_primary,$opt_ssh,$opt_sshd,$opt_ntp,$opt_sudo,$opt_hostname,
                             $opt_force,$options,'--unattended']), ''))
 
       exec { 'ipa_installer':
diff --git a/manifests/params.pp b/manifests/params.pp
index 0b77051..e1fa1a6 100644
--- a/manifests/params.pp
+++ b/manifests/params.pp
@@ -15,6 +15,7 @@ class ipaclient::params {
   $automount      = false
   $mkhomedir      = true
   $sudo           = true
+  $hostname      = $::fqdn
   $fixed_primary  = false
   $options        = ''
   $installer      = '/usr/sbin/ipa-client-install'
```

**Comment by joshuabaird** [comment]
@brightdroid, thanks for the contribution!  Could you please open a PR?

**Comment by joshuabaird** [comment]
@brightdroid, wondering what version of ``ipa-client-install`` you are using and on which distribution?  Can you show the error that is thrown if you don't have a FQDN in ``/etc/hostname``?  I want to make sure this is the expected behavior.

**Comment by joshuabaird** [comment]
I just ran a quick test, and ``ipa-client-install`` did not fail for me with a short-name set as my hostname:

```
root@ocpdv-d1-ms01:~# hostname
ocpdv-d1-ms01
root@ocpdv-d1-ms01:~# cat /etc/hostname
ocpdv-d1-ms01
root@ocpdv-d1-ms01:~# ipa-client-install --no-ssh --no-sshd --no-ntp --domain=qa-unix.follett.com --mkhomedir
Discovery was successful!
Client hostname: ocpdv-d1-ms01.corp.follett.com
Realm: QA-UNIX.FOLLETT.COM
DNS Domain: qa-unix.follett.com
IPA Server: imqa-d1-dc02.qa-unix.follett.com
BaseDN: dc=qa-unix,dc=follett,dc=com

Continue to configure the system with these values? [no]: yes
Skipping synchronizing time with NTP server.
User authorized to enroll computers: jbairdipa
Password for jbairdipa@QA-UNIX.FOLLETT.COM:
Successfully retrieved CA cert
    Subject:     CN=Certificate Authority,O=QA-UNIX.FOLLETT.COM
    Issuer:      CN=Certificate Authority,O=QA-UNIX.FOLLETT.COM
    Valid From:  Tue Nov 08 18:04:50 2016 UTC
    Valid Until: Sat Nov 08 18:04:50 2036 UTC

Enrolled in IPA realm QA-UNIX.FOLLETT.COM
Created /etc/ipa/default.conf
New SSSD config will be created
Configured sudoers in /etc/nsswitch.conf
Configured /etc/sssd/sssd.conf
Configured /etc/krb5.conf for IPA realm QA-UNIX.FOLLETT.COM
trying https://imqa-d1-dc02.qa-unix.follett.com/ipa/json
Forwarding 'ping' to json server 'https://imqa-d1-dc02.qa-unix.follett.com/ipa/json'
Forwarding 'ca_is_enabled' to json server 'https://imqa-d1-dc02.qa-unix.follett.com/ipa/json'
Systemwide CA database updated.
Added CA certificates to the default NSS database.
Adding SSH public key from /etc/ssh/ssh_host_rsa_key.pub
Adding SSH public key from /etc/ssh/ssh_host_ecdsa_key.pub
Adding SSH public key from /etc/ssh/ssh_host_ed25519_key.pub
Forwarding 'host_mod' to json server 'https://imqa-d1-dc02.qa-unix.follett.com/ipa/json'
Could not update DNS SSHFP records.
SSSD enabled
Configured /etc/openldap/ldap.conf
Configuring qa-unix.follett.com as NIS domain.
Client configuration complete.
```

This is on EL7.2 and ``ipa-client-4.2.0-15.el7_2.19.x86_64``.  What switches/version/dist are you using when you see the failure?

**Comment by joshuabaird** [comment]
Additionally, there is some discussion on freeipa-users about changing the behavior of ``--hostname`` so it doesn't actually modify the hostname of the system:

https://www.redhat.com/archives/freeipa-users/2016-December/msg00089.html

I'm not quite sure if this changes anything with the expected behavior or not.

**Comment by joshuabaird** [comment]
Closed/fixed in favor of #53.
